src/sdk/pynni/nni/README.md

```diff
@@ -2,92 +2,183 @@
+This optimization approach is described in detail in [Algorithms for Hyper-Parameter Optimization][1].
     
-The TPE approach models P(x|y) and P(y) where x represents hyperparameters and y the associated evaluate matric. P(x|y) is modeled by transforming the generative process of hyperparameters, replacing the distributions of the configuration prior with non-parametric densities. This optimization approach is described in detail in [Algorithms for Hyper-Parameter Optimization][1].
-    
-Comparing with other algorithm, TPE could be achieve better result when the number of trial experiment is small. Also TPE support continuous or discrete hyper-parameters. From a large amount of experiments, we could found that TPE is far better than Random Search.
-
+_Suggested scenario_: TPE, as a black-box optimization, can be used in various scenarios, and shows good performance in general. Especially when you have limited computation resource and can only try a small number of trials. From a large amount of experiments, we could found that TPE is far better than Random Search.
+
+_Usage_:
+```
+  # config.yaml
+  tuner:
+    builtinTunerName: TPE
+    classArgs:
+      # choice: maximize, minimize
+      optimize_mode: maximize
+```
+
+<a name="Random"></a>
 **Random Search**
 
 In [Random Search for Hyper-Parameter Optimization][2] show that Random Search might be surprisingly simple and effective. We suggests that we could use Random Search as baseline when we have no knowledge about the prior distribution of hyper-parameters.
-    
+
+_Suggested scenario_: Random search is suggested when each trial does not take too long (e.g., each trial can be completed very soon, or early stopped by assessor quickly), and you have enough computation resource. Or you want to uniformly explore the search space.
+
+_Usage_:
+```
+  # config.yaml
+  tuner:
+    builtinTunerName: Random
+```
+
+<a name="Anneal"></a>
 **Anneal**
-    
+
+Intro here.
+
+_Suggested scenario_:
+
+_Usage_:
+
+<a name="Evolution"></a>
 **Naive Evolution**
 
-Naive Evolution comes from [Large-Scale Evolution of Image Classifiers][3]. Naive Evolution require more experiments to works, but it's very simple and easily to expand new features. There are some tips for user: 
+Naive Evolution comes from [Large-Scale Evolution of Image Classifiers][3]. It randomly initializes a population based on search space. For each generation, it chooses better ones and do some mutation (e.g., change a hyperparameter, add/remove one layer) on them to get the next generation. 
+
+_Suggested scenario_: Naive Evolution also requires many trials to works. Specifically, it requires large inital population to avoid falling into local optimum. This tuner is more suggested when your trial code supports weight transfer, that is, the trial could inherit the converged weights from its parent(s). This can greatly speed up the training progress.

```

Comment:
Naive Evolution require more experiments to works, but it's very simple and easily to expand new features.   please maintain whole sentence.

Reply:
thanks, will update.